ProcessMaker/processmaker
# FOUR-13345 Block duplicate and Kill existing session in Device Restriction is not working

## Issue & Reproduction Steps
- Block duplicate session in Device Restriction is not working
- Kill existing session in Device Restriction is not working

## Solution
- Fix session block middleware
- Fix user session listener

## How to Test
- Block duplicate session
1. Log in with admin user
2. Configure `Block duplicate session` in Device Restriction
   1. Go to Admin tab
   2. Go to settings option in sidebar menu
   3. Go to Session Control tab
   4. Go to Device Restriction and edit
   5. Enable Block duplicate session
   6. Click on save button
3. Create a new user
4. Login with new user in a first device
5. Login with new user in a second device
6. Only the first session should remain open, the second session should be blocked

- Kill existing session
1. Log in with admin user
2. Configure `Kill existing session` in Device Restriction
   1. Go to Admin tab
   2. Go to settings option in sidebar menu
   3. Go to Session Control tab
   4. Go to Device Restriction and edit
   5. Enable Kill existing session
   6. Click on save button
3. Create a new user
4. Login with new user in a first device
5. Login with new user in second device
6. Only one session should be open, not both

## Related Tickets & Packages
[FOUR-13345](https://processmaker.atlassian.net/browse/FOUR-13345)
[FOUR-13347](https://processmaker.atlassian.net/browse/FOUR-13347)

## Code Review Checklist

ci:next
ci:deploy

[FOUR-13345]: https://processmaker.atlassian.net/browse/FOUR-13345?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
